tests/bugfixes/github/test_issue_981.py

```diff
@@ -32,7 +32,7 @@ class CanonAfInfoTest(metaclass=CaseMeta):
 Exif.Canon.AFPointsInFocus                   Short       1  4
 Exif.Canon.AFPointsSelected                  Short       1  8
 Exif.Canon.AFPointsUnusable                  Short       1  (none)
-Exif.Canon.AFInfo                            Short     273  546 2 63 61 6720 4480 6720 4480 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 0 0 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 0 0 65200 64790 64435 64099 63764 336 0 65200 64099 63764 1772 1437 1101 746 336 0 1437 1101 746 336 0 65200 64790 64435 336 0 65200 64790 64435 64099 63764 1772 64790 64435 64099 63764 1772 1437 1101 746 63764 1772 1437 1101 746 336 0 65200 1101 746 336 0 65200 64790 64435 64099 336 0 65200 1772 1437 0 0 547 625 625 625 625 821 821 821 308 308 625 625 625 625 547 547 308 308 308 274 274 274 308 308 0 0 0 0 0 0 0 308 65228 65228 65228 65228 0 0 0 0 64911 65228 65228 65228 65228 65262 65262 65262 64911 64911 64989 64989 64989 64911 64911 64911 64715 64715 64715 64911 64911 0 0 0 512 0 0 0 512 0 0 0 0 0 0 65535
+""" , """Exif.Canon.AFInfo                            Short     273  546 2 63 61 6720 4480 6720 4480 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 0 0 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 218 0 0 65200 64790 64435 64099 63764 336 0 65200 64099 63764 1772 1437 1101 746 336 0 1437 1101 746 336 0 65200 64790 64435 336 0 65200 64790 64435 64099 63764 1772 64790 64435 64099 63764 1772 1437 1101 746 63764 1772 1437 1101 746 336 0 65200 1101 746 336 0 65200 64790 64435 64099 336 0 65200 1772 1437 0 0 547 625 625 625 625 821 821 821 308 308 625 625 625 625 547 547 308 308 308 274 274 274 308 308 0 0 0 0 0 0 0 308 65228 65228 65228 65228 0 0 0 0 64911 65228 65228 65228 65228 65262 65262 65262 64911 64911 64989 64989 64989 64911 64911 64911 64715 64715 64715 64911 64911 0 0 0 512 0 0 0 512 0 0 0 0 0 0 65535
```

Comment:
👀 how is possible that this was working previously?

Reply:
Yes.  The Canon.AFxxx behaviour is unchanged since 0.27.3.  I modified the test slightly because a couple of new fields were add `Exif.Canon.AFMicroAdj` and `Exif.Canon.AFConfig`.  I don't know the genesis of these fields.  @piponazo @alexvanderberkel @hassec and @neheb get a mention in the blame listing:

**AFConfig**
```bash
534 rmills@rmillsm1:~/gnu/github/exiv2/main $ git blame src/canonmn_int.cpp | grep AFConfig 
bdd8a386b src/canonmn_int.cpp (Christoph Hasse    2021-06-08 21:56:04 +0200  637)         {0x4028, "AFConfig", N_("AFConfig"), N_("AFConfig"), canonId, makerTags, unsignedShort, -1, printValue},
ff2ffb190 src/canonmn_int.cpp (Alex Esseling      2021-01-12 20:55:29 +0100 1230)     //Canon AFConfig Tags
6da49fd29 src/canonmn_int.cpp (Rosen Penev        2021-05-17 16:54:53 -0700 1232)          {0x0001, "AFConfigTool", N_("AF Config Tool"), N_("AF Config Tool"), canonAfCId, makerTags, signedLong, -1, printValue},
```

**AFMicroAdj**
```
535 rmills@rmillsm1:~/gnu/github/exiv2/main $ git blame src/canonmn_int.cpp | grep AFMicroAdj
bdd8a386b src/canonmn_int.cpp (Christoph Hasse    2021-06-08 21:56:04 +0200  628)         {0x4013, "AFMicroAdj", N_("AFMicroAdj"), N_("AFMicroAdj"), canonId, makerTags, unsignedShort, -1, printValue},
ff2ffb190 src/canonmn_int.cpp (Alex Esseling      2021-01-12 20:55:29 +0100  906)     // Canon AFMicroAdjMode Quality Info, tag 0x0001
6da49fd29 src/canonmn_int.cpp (Rosen Penev        2021-05-17 16:54:53 -0700  907)     constexpr TagDetails canonAFMicroAdjMode[] = {
bdd8a386b src/canonmn_int.cpp (Christoph Hasse    2021-06-08 21:56:04 +0200  914)      // Canon AFMicroAdj Info Tag
6da49fd29 src/canonmn_int.cpp (Rosen Penev        2021-05-17 16:54:53 -0700  916)          {0x0001, "AFMicroAdjMode", N_("AFMicroAdjMode"), N_("AFMicroAdjMode"), canonAfMiAdjId, makerTags, signedLong, -1, EXV_PRINT_TAG(canonAFMicroAdjMode)},
bdd8a386b src/canonmn_int.cpp (Christoph Hasse    2021-06-08 21:56:04 +0200  917)          {0x0002, "AFMicroAdjValue", N_("AF Micro Adj Value"), N_("AF Micro Adj Value"), canonAfMiAdjId, makerTags, signedRational, -1, printValue},
de036b2b2 src/canonmn_int.cpp (Luis Díaz Más      2022-02-09 23:00:03 +0100  918)          {0xffff, "(UnknownCanonAFMicroAdjTag)", "(UnknownCanonAFMicroAdjTag)", N_("Unknown Canon AFMicroAdj tag"), canonAfMiAdjId, makerTags, signedShort, 1, printValue}
536 rmills@rmillsm1:~/gnu/github/exiv2/main $ 
 ```